src/prototype/main.py

```diff
@@ -0,0 +1,85 @@
+import sys
+import re
+import os
+from PyQt5.QtWidgets import QApplication, QMainWindow
+from PyQt5.QtCore import pyqtSlot
+from gui.mainwidow_ui import Ui_MainWindow
+from OdasStream.odas_stream import OdasStream
+from FileHelper.file_helper import FileHelper
+from ArgsParser.args_parser import ArgsParser
+
+
+class MainWindow(QMainWindow, Ui_MainWindow):
+    def __init__(self, odasStream, parent=None):
+        super(MainWindow, self).__init__(parent)
+        self.setupUi(self)
+        self.odasStream = odasStream
+
+        # signal slots
+        self.odasStream.signalOdasData.connect(self.odasDataReveived)
+        self.btnStartOdas.clicked.connect(self.btnStartOdasClicked)
+        self.btnStopOdas.clicked.connect(self.btnStopOdasClicked)
+
+
+    def closeEvent(self, event):
+        self.stopOdas()
+        event.accept()
+
+
+    def startOdas(self):
+        if self.odasStream and not self.odasStream.isRunning:
+            self.odasStream.start()
+
+
+    def stopOdas(self):
+        if self.odasStream and self.odasStream.odasProcess:
+            if self.odasStream.isRunning:
+                self.odasStream.stop()
+
+
+    @pyqtSlot()
+    def btnStartOdasClicked(self):
+        self.startOdas()
+
+
+    @pyqtSlot()
+    def btnStopOdasClicked(self):
+        self.stopOdas()
+
+
+    @pyqtSlot(object)
+    def odasDataReveived(self, values):
+        self.source1AzimuthValue.setText('%.5f' % values[0]['azimuth'])
+        self.source2AzimuthValue.setText('%.5f' % values[1]['azimuth'])
+        self.source3AzimuthValue.setText('%.5f' % values[2]['azimuth'])
+        self.source4AzimuthValue.setText('%.5f' % values[3]['azimuth'])
+
+        self.source1ElevationValue.setText('%.5f' % values[0]['elevation'])
+        self.source2ElevationValue.setText('%.5f' % values[1]['elevation'])
+        self.source3ElevationValue.setText('%.5f' % values[2]['elevation'])
+        self.source4ElevationValue.setText('%.5f' % values[3]['elevation'])
```

Comment:
We should initialize all class attributes in __init__ to regroup them. After that, instead of 1 attribute for each source we should use a dictionary named self.sourcesValues. in that dictionary you'll have 2 keys, which "azimuth" and "elevation".  Those keys will be link to sub dictionaries, which will contain a key for each sources and the value associted:

self.sourcesValues = 
{
    'azimuth' : {
        '0' : value,
        '1' : value,
        '2' : value,
        '3' : value,
    },
    'elevation' : {
        '0' : value,
        '1' : value,
        '2' : value,
        '3' : value,
    },
}

Reply:
We will talk about it